File: front/front-monopoly/src/services/GameRoomService.js
import axios from 'axios';

const GAMEROOM = 'http://localhost:9090/api/gamerooms';

class GameRoomService {
    getAllGameRooms(){
        return axios.get(GAMEROOM + '/getAllGameRooms')
    }

    createGameRoom(gameRoom) {
        return axios.post(GAMEROOM + '/addGameRoom', gameRoom); 
    }

    updateGameRoomById(gameRoomId, gameRoom) {
        return axios.post(GAMEROOM + '/updateGameRoomById/' + gameRoomId, gameRoom);
    }

    getGameRoomById(gameRoomId) {
        return axios.get(GAMEROOM + '/getGameRoomById/' + gameRoomId);
    }
}

export default new GameRoomService();

File: src/main/java/aeh/zzaip/Monopoly/service/GameRoomService.java
package aeh.zzaip.Monopoly.service;

import aeh.zzaip.Monopoly.model.GameRoom;
import aeh.zzaip.Monopoly.model.Player;

import java.util.List;
import java.util.Optional;

public interface GameRoomService {
    List<GameRoom> getAllGameRooms();

    Optional<GameRoom> getGameRoomById(Long gameRoomId);

    GameRoom addGameRoom(GameRoom gameRoom);

    Optional<GameRoom> updateGameRoomById(Long gameRoomId, GameRoom newGameRoom);

    boolean deleteGameRoomById(Long gameRoomId);
}
